apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx

```diff
@@ -0,0 +1,697 @@
+
+  const { isOpen: isOpenCombobox, close: closeCombobox, open: openCombobox, toggle: toggleCombobox } = useDisclosure();
+  const { refs: comboboxRefs, style: comboboxStyle } = useDropdown({
+    isOpen: isOpenCombobox,
+    onClose: closeCombobox,
+    placement: 'bottom-start',
+    middleware: [offset(4)],
+  });
+  const comboboxId = useId();
+  const comboboxListId = useId();
+
+  const {
+    current: currentFocus,
+    focusables: focusableElements,
+    updateFocusableElements,
+  } = useTrapFocus(comboboxDropdownRef, {
+    trapTabs: false,
+    arrowKeysUpDown: true,
+    activeState: isOpenCombobox,
+    initialFocus: false,
+  });
+
+  const handleFocusInput = () => {
+    comboboxInputRef.current?.focus();
+  };
+
+  const handleReset = () => {
+    setSearchValue('');
+    setSnippets([]);
+    closeCombobox();
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const phrase = event.target.value;
+    setSelectedValueCombobox('');
+    if (phrase) {
+      setSearchValue(phrase);
+    } else {
+      handleReset();
+    }
+  };
+
+  const handleInputKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === 'Escape') handleReset();
+    if (event.key === 'Enter') closeCombobox();
+    if (event.key === 'ArrowUp') {
+      openCombobox();
+      updateFocusableElements();
+      if (isOpenCombobox && focusableElements.length > 0) {
+        focusableElements[focusableElements.length - 1].focus();
+      }
+    }
+    if (event.key === 'ArrowDown') {
+      openCombobox();
+      updateFocusableElements();
+      if (isOpenCombobox && focusableElements.length > 0) {
+        focusableElements[0].focus();
+      }
+    }
+  };
+
+  const selectOptionCombobox = (event: FormEvent, option: SelectOption) => {
+    setSearchValue(option.label);
+    setSelectedValueCombobox(option.label);
+    closeCombobox();
+    handleFocusInput();
+  };
+
+  const handleOptionItemKeyDownCombobox = (event: KeyboardEvent<HTMLButtonElement>, option: SelectOption) => {
+    if (event.key === 'Escape') {
+      handleFocusInput();
+    } else if (event.key === ' ' || event.key === 'Enter') selectOptionCombobox(event, option);
+  };
+
+  const mockAutocompleteRequest = (phrase: string) => {
+    const results = options.filter((option) => option.value.toLowerCase().startsWith(phrase.toLowerCase()));
+    return results;
+  };
+
+  useEffect(() => {
+    if (searchValue && !selectedValueCombobox) {
+      const getSnippets = async () => {
+        openCombobox();
+        try {
+          const data = await mockAutocompleteRequest(searchValue);
+          setSnippets(data);
+        } catch (error) {
+          closeCombobox();
+          // eslint-disable-next-line no-console
+          console.error(error);
+        }
+      };
+      getSnippets();
+    }
+  }, [searchValue]);
+
+  const { close, toggle, isOpen } = useDisclosure({ initialValue: false });
```

Comment:
```suggestion
  const { close, toggle, isOpen } = useDisclosure();
```
`initialValue: false` is default

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2787/commits/4340777142524b2132252c34501b858df32e129f